lib/spack/spack/oci/oci.py

```diff
@@ -69,6 +69,39 @@ def with_query_param(url: str, param: str, value: str) -> str:
     )
 
 
+def list_tags(ref: ImageReference) -> List[str]:
+    """Retrieves the list of tags associated with an image
+
+    Fetches all tags from an image's tags_url, handling pagination of
+    results.
+
+    Args:
+        reg: The image reference.
+    Returns:
+        List of strings representing all tags associated with the image.
+
+    """
+    tags = []
+    fetch_url: Optional[str] = ref.tags_url()
+
+    while fetch_url:
+        request = Request(url=fetch_url)
+        response = spack.oci.opener.urlopen(request)
+        spack.oci.opener.ensure_status(request, response, 200)
+        tags.extend(json.load(response)["tags"])
+        fetch_url = None
+        link_header = spack.util.web.get_header(response.headers, "Link")
```

Comment:
This is not part of the spec.

Reply:
> This is not part of the spec.

Do you have a link for reference?  The link I pasted in the description seems to be for oci distribution v1.0.1, and it has a section that reads:

```
<name> is the namespace of the repository, <int> is the number of tags requested, and <tagname> is the value of the last tag. <tagname> MUST NOT be a numerical index, but rather it MUST be a proper tag. A request of this sort will return up to <int> tags, beginning non-inclusively with <tagname>. That is to say, <tagname> will not be included in the results, but up to <int> tags after <tagname> will be returned. The tags MUST be in lexical order.
```

Though I was confused by some sections below that seem to contradict what `tagname>` is supposed to be.  Am I looking at completely the wrong spec?

Reply:
The spec does not say anything about how pagination through Link headers, I think that's just how gitlab's registry does it